### PR TITLE
feat(api): add E2EI and ACME APIs [AR-1957]

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Add ACME server and Wire E2EI APIs to network.

In this PR are 2 APIs classes, one for E2EI against wire server, and one for ACME server.
The E2EIApi needs authentication so I put in the Authenticated layer; but for the ACMEApi it's not wire server and there is no need for authentications or wire headers.

Also moved ignoreSSLCertificates option only to unbound network layer for safety and make sure we have it only in the place we need it.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v4/authenticated/E2EIApiV4.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            handleNonceResponse(httpResponse)
+        }
+
+    private suspend fun handleNonceResponse(
+        httpResponse: HttpResponse
+    ): NetworkResponse<String> = if (httpResponse.status.isSuccess())
+        httpResponse.headers[NONCE_HEADER_KEY]?.let { nonce ->
+            NetworkResponse.Success(nonce, httpResponse)
+        } ?: run {
+            CustomErrors.MISSING_NONCE
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v4/authenticated/E2EIApiV4.kt:52]: can we please add test to this case

**Review by MohamadJaara — APPROVED** [review]
LGTM, only small comment about missing test

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v4/authenticated/E2EIApiV4.kt (hunk 1 above)
Comment (mchenani) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/v4/authenticated/E2EIApiV4.kt:52]: Testing are working in progress, but sure will cover this with Priority.

Comment (mchenani) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/v4/authenticated/E2EIApiV4.kt:52]: Test added.

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/base/unbound/acme/ACMEApi.kt [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    suspend fun getNewOrder(url: String, body: ByteArray): NetworkResponse<ACMEResponse>
+
+    suspend fun dpopChallenge(url: String, body: ByteArray): NetworkResponse<ChallengeResponse>
+
+    suspend fun oidcChallenge(url: String, body: ByteArray): NetworkResponse<ChallengeResponse>
+
+    suspend fun getAuthzChallenge(url: String): NetworkResponse<ACMEResponse>
+
+    suspend fun getAuhzDirectories(): NetworkResponse<AuthzDirectoriesResponse>
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/base/unbound/acme/ACMEApi.kt:55]:
```suggestion
    suspend fun getAuthzDirectories(): NetworkResponse<AuthzDirectoriesResponse>
```

**Review by typfel — CHANGES_REQUESTED** [review]
Looking good, except for one typo.

Tests for the ACMEApi will come in a follow up PR?

**Comment by mchenani** [comment]
> Looking good, except for one typo.
> 
> Tests for the ACMEApi will come in a follow up PR?

The typo fixed.
Tests are coming in next PRs.

**Review by typfel — APPROVED** [review]

**Pull request merged**